### ISSUE TITLE
Find alternative ways to tag and update inserted statistics

### ISSUE THREAD

**Issue by WillemSleegers** [gasparl/tidystats-Google-Docs-add-in#4]
The current solution that inserts statistics as links works (yay), but has some limitations:
1. The numbers have a different color compared to the main text (probably easily fixable)
2. Hovering on them reveals a partial link, which is distracting because it's not useful, and the link just takes you to the website, which is also not useful

I wonder what alternative methods are possible and if some of them are potentially better (and less hacky).

**Comment by gasparl** [comment]
The color issue is now fixed.

For the link, as discussed, there seems to be no reasonable alternative. The link could look like as `tidystats.io/?id=...`, where `...` is the key for the given statistic, e.g.: `tidystats.io/?id=t_test_paired$estimate`. Then this link would redirect to the [tidystats.io](tidystats.io) website, where a short explanation may be given that this is just part of the tidystats mechanism. Similarly, in the sidebar we could explain this shortly, e.g. as follows.

> All statistical numbers and other values inserted via _tidystats_ are "tagged" with a hyperlink. This hyperlink contains the identifier of the given statistic, and it thereby remains updatable: when you upload a new file with changed statistical values for the same test, all values tagged with such identifying _tidystats_ links in the document will be updated with the corresponding new values (if any) for the same statistics from the file. If you want the values to remain updatable, keep the link intact! (If you no longer wish them to be updatable, you can freely remove the links.)

In case of the website, you can also add a note that the specific identifier refers to such and such statistics. For instance, in the above example, it could say that "This specific identifier (`t_test_paired$estimate`) points to the `estimate` value of `t_test_paired`."

**Comment by WillemSleegers** [comment]
I've been playing around with this a bit and realized that when you insert a normal link in Google Docs (not tidystats related) it actually looks pretty good:
<img width="339" alt="Screen Shot 2022-06-17 at 08 55 37" src="https://user-images.githubusercontent.com/4398042/174242920-61e81a02-f74a-4210-9479-c0adbabc4655.png">

I've tried to get the same result by linking to `tidystats.io` and see if it gets a nice link preview like that, but I haven't been successful yet. Maybe it's something you can look into as well.

So far I've tried adding meta title and description tags to the website to get the preview, but no luck yet. It could be that it takes some time for Google to crawl the website and so it will only show up later, but not sure if I'm right about this.

EDIT: It does work. If I link to http://tidystats.io, I get a preview of the website. I think we should do that, and then I have no problem with how it looks. =)

EDIT: And by 'that', I mean we should link to a website and provide some useful information so the preview looks informative and non-scary.

**Comment by gasparl** [comment]
The problem is that this only works with root hyperlinks such as [https://www.google.com/](https://www.google.com/). If you add any query string like [https://www.google.com/?id=sth](https://www.google.com/?id=sth), or any other modification, the preview is gone.

But I'll look into it and get back to you.

**Comment by WillemSleegers** [comment]
Something like https://www.tidystats.io/word-add-in.html#test=2 seems to work though.

**Comment by gasparl** [comment]
Yep, I'm experimenting with the hashtag solution at the moment and see if there is any drawback.

**Comment by gasparl** [comment]
Everything seems fine. It's cool: you can probably also include some more relevant information in the bubble's text too. (With this we are ahead of Zotero ;) )

So it now looks like e.g. `tidystats.io/#id=stat$value`. For brevity/simplicity I suggest keeping this short format (without /sth.html); then you can just [check the hash values](https://stackoverflow.com/questions/298503/) e.g. on page load and correspondingly show a simple (and closable) information `div` (like an in-page popup – but you can also redirect to another page if you wish).

(If for some reason it's necessary to add another page, e.g. because you want a different icon and description, maybe you could do something like `tidystats.io/tag` or such; and then just put the necessary info in the corresponding `index.html`.)

**Comment by WillemSleegers** [comment]
Great, that seems pretty good.

The only thing I'm unsure about is how Google figures out the title and description; so I'm not yet entirely sure how to code the page.

**Comment by gasparl** [comment]
The title is ~clearly~ taken from `<title>tidystats</title>`. (I checked and it's the same for other sites.)

However, the image and text preview seems to be a mystery: see e.g. [here](https://stackoverflow.com/questions/62728331/) and [here](https://support.google.com/docs/thread/22081226?hl=en). Interestingly, GDocs did find your tidystats (large) logo for preview. One hacky trick would be to change this logo to include the text that you want people to see when they click on the link-tagged values. (And make it invisible on the website if you don't want it displayed.) But we can never be certain without official documentation, which seems not to exist so far.

Maybe we could put a bounty on [this](https://stackoverflow.com/questions/62728331/) SO question, just in case there is some latest update we missed.

**Comment by WillemSleegers** [comment]
The title is not that clear though. If I link to the Word add-in page on tidystats.io, the title becomes "Word add-in", which is not in the title tag of the HTML.

**Comment by gasparl** [comment]
Ah okay, then it's really unpredictable, because I checked several sites where the `title` tag was taken.

**Comment by WillemSleegers** [comment]
I've been trying to figure this out and apparently Google has their own technique for creating the title based on various parts of the page. This is extremely frustrating and another reason to be annoyed at Google.
